src/cryptenroll/cryptenroll-tpm2.c

```diff
@@ -148,6 +150,8 @@ int enroll_tpm2(struct crypt_device *cd,
         ssize_t base64_encoded_size;
         int r, keyslot;
         TPM2Flags flags = 0;
+        uint8_t binary_salt[SHA256_DIGEST_SIZE] = {};
+        CLEANUP_ERASE(binary_salt);
```

Comment:
the salt is not a secret, hence CLEANUP_ERASE() is not necessary (and misleading) here, no?

Reply:
Not a secret, but right now to get the value the attacker would have to get it from the luks header over perhaps some core dump. If the attacker has it, they could start to use it to hammer a recorded encrypted session, this actually gets fixed when I move your code base to the SRK, since they can't fake your primary key. Ie right now, someone could be rogue on the bus and giving you an attacker controlled primary key.

I was thinking it was prudent to cleanse the salt out to keep it out of core dumps, but I won't die on that hill.

Reply:
@poettering the PR has been updated, this is the only change I didn't make, let me know if you would still like me to make it. Thank you.

Reply:
maybe add a brief explanation as a comment then

Reply:
@poettering done